diff '--color=auto' -ruNp a/dwm.c b/dwm.c
--- a/dwm.c	2021-10-29 10:24:01.480633049 +0200
+++ b/dwm.c	2021-10-29 10:25:37.257765408 +0200
@@ -876,7 +876,7 @@ drawtab(Monitor *m) {
 	m->ntabs = 0;
 	for(c = m->clients; c; c = c->next){
 	  if(!ISVISIBLE(c)) continue;
-	  m->tab_widths[m->ntabs] = TEXTW(c->name);
+	  m->tab_widths[m->ntabs] = MIN(TEXTW(c->name), 250);
 	  tot_width += m->tab_widths[m->ntabs];
 	  ++m->ntabs;
 	  if(m->ntabs >= MAXTABS) break;
